pdg/src/info.rs

```diff
@@ -175,65 +175,97 @@ mod test {
     use c2rust_analysis_rt::mir_loc::Func;
     use rustc_middle::mir::Local;
 
-    fn mk_node(g: &mut Graph, kind: NodeKind, source: Option<NodeId>) -> NodeId {
-        g.nodes.push(Node {
-            function: Func {
-                def_path_hash: (1, 2).into(),
-                name: "fake_function".into(),
-            },
-            block: 0_u32.into(),
-            statement_idx: 0,
-            dest: None,
-            kind,
-            source,
-            info: None,
-            debug_info: "".into(),
-        })
-    }
-
-    fn mk_addr_of_local(g: &mut Graph, local: impl Into<Local>) -> NodeId {
-        mk_node(g, NodeKind::AddrOfLocal(local.into()), None)
-    }
-
-    fn mk_copy(g: &mut Graph, source: NodeId) -> NodeId {
-        mk_node(g, NodeKind::Copy, Some(source))
-    }
-
-    fn mk_store_addr(g: &mut Graph, source: NodeId) -> NodeId {
-        mk_node(g, NodeKind::StoreAddr, Some(source))
-    }
+    #[derive(Debug)]
+    struct CheckedNode {
+        id: NodeId,
+        unique: bool,
+        stmt: &'static str,
+    }
+
+    #[derive(Debug, Default)]
+    struct CheckedGraph {
+        graph: Graph,
+        checked_nodes: Vec<CheckedNode>,
+    }
+
+    impl CheckedGraph {
+        fn mk_node(
+            &mut self,
+            stmt: &'static str,
+            unique: bool,
```

Comment:
nit pick: i think i'd rather this be an enum than a bool

Reply:
optional change, just my two cents

Reply:
What's the advantage of an `enum`?  If we're calling them `Unique` and `NonUnique`, I'm not sure there's an advantage over a `bool`.  One is just non the other.  Plus, then we should just change the `unique` field in `NodeInfo::unique` to the same `enum`.

Reply:
> Plus, then we should just change the unique field in NodeInfo::unique to the same enum.

I agree that should be done. My preference is because I don't like relying on IDEs to tell me what the bool represents. A type `Unique` or `NonUnique` is very clear, despite it being semantically the same as a `bool`. In cases where functions take more than one bool (not the case here), it prevents mixing argument order by making things type safe.

Reply:
Okay, that makes sense.  Plus we could change it to `Unique` and `Aliases` if we want, too.  I do think we should save that for a later PR, though.

Reply:
that sounds okay